modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -1507,6 +1507,7 @@ CompletableFuture<Ignite> joinClusterAsync() {
                 }, joinExecutor)
                 .thenComposeAsync(ignored -> awaitSelfInLocalLogicalTopology(), joinExecutor)
                 .thenCompose(ignored -> catalogManager.catalogInitializationFuture())
+                .thenCompose(ignored -> systemViewManager.completeRegistration())
```

Comment:
This is how it was originally implemented, but it causes a noticeable slowdown in running runAll tests, so it was decided to remove it.